docs/contributor-guide.md

```diff
@@ -30,22 +30,181 @@ You can see which steps will be performed by running 'dh binary --no-act' in the
+Actual storage data is put to `/var/lib/storagenode/storage`.
+
+#### `override_dh_installsystemd`
+```make
+override_dh_installsystemd:
+	dh_installsystemd --name=storagenode
+	dh_installsystemd --name=storagenode-updater
+```
+
+Overrides default `override_dh_installsystemd` to install both `storagenode` and `storagenode-updater` services.
+
+Be careful, rules is not the right place to make changes to the user config. These steps are not run on the user machine.
+They are only performed when building the package.
+
+### `preinst`
+This is the script that will be run before the installation process has completed.
+It takes as an argument the operation that is being performed ('install', 'install <old-version>', 'upgrade <new-version>',
+'abort-upgrade <new-version>').
+
+- Check if host architecture is supported. `supported archs`: `amd64 armhf arm64`
 
 ### `postinst`
-This is the script that will be run after the installation process has completed. It takes as an argument the operation that is being performed ('configure', 'abort-upgrade', 'abort-remove', 'abort-deconfigure').
+This is the script that will be run after the installation process has completed.
+It takes as an argument the operation that is being performed ('configure', 'abort-upgrade', 'abort-remove', 'abort-deconfigure').
 
 In our case, we perform the following operations:
 - retrieve the user configuration using debconf
-- download the binaries
+- download the binaries for host architecture
 - add a user in the system that will run the services
+- copy identity to `/etc/storagenode/identity` if none
+- setup storagenode if it hasn't been setup yet
+- change ownership of `/var/lib/storagenode` and `/etc/storagenode` directories
 
 ### `prerm`
-This file contains the steps to be run when removing the package.
+This file contains the steps to be run before removing the package.
+
+### `postrm`
+This file contains the steps to be run after removing the package. Removes storagenode and storagenode-updater binaries.
 
 ### `storagenode.service` and `storagenode-updater.service`
 These are the services that will be installed and which will run the binaries.
 
+#### `storagenode` service unit file
+```service
+# This is a SystemD unit file for the Storage Node
+# To configure:
+# - Update the user and group that the service will run as (User & Group below)
+# - Ensure that the Storage Node binary is in /usr/local/bin and is named storagenode (or edit the ExecStart line
+#   below to reflect the name and location of your binary
+# - Ensure that you've run setup and have edited the configuration appropriately prior to starting the
+#   service with this script
+# To use:
+# - Place this file in /etc/systemd/system/ or wherever your SystemD unit files are stored
+# - Run systemctl daemon-reload
+# - To start run systemctl start storagenode
+
+[Unit]
+Description  = Storage Node service
+After        = syslog.target
```

Comment:
I had problems with this on my debian system. Sometimes the storage node was starting before the network interface was ready. For that reason my service is using:
```
After        = network-online.target
Wants        = network-online.target
```

Reply:
Yes! We definitely need the network stack to be up. I think, though, that we want instead only one line:

    After = syslog.target network.target

It's recommended as good practice that most network servers use `network.target` instead of `network-online.target`, since servers can work without an externally routable IP (using, for example, localhost). And no `Wants` is needed for `network.target` because it is a passive unit.